src/relax/transform/fuse_ops.cc

```diff
@@ -262,16 +262,11 @@ class GraphCreator : public ExprVisitor {
     IndexedForwardGraph::Node* leaf_node = nullptr;
     if (it != graph_.node_map.end()) {
       leaf_node = it->second;
-    } else if (leaf_expr->IsInstance<ConstantNode>() || leaf_expr->IsInstance<ShapeExprNode>() ||
-               leaf_expr->IsInstance<PrimValueNode>() || leaf_expr->IsInstance<StringImmNode>() ||
-               leaf_expr->IsInstance<DataTypeImmNode>()) {
+    } else {
```

Comment:
@tqchen Responding to the comment [here](https://github.com/apache/tvm/pull/17022#discussion_r1613292617), with conversation migrated to the new PR.

> Is this intended change? Seems that we should consider sass where constant value(and there may not be a binding pt)

This change is intentional, and maintains the same behavior for constants as before.  Prior to this change, the `else if` condition was entered for five of the six child classes of `relax::LeafExprNode`, and the `else` branch with the `LOG(FATAL)` was only entered for a `relax::Var`.

With this change, the behavior of all LeafExpr subclasses is the same: If encoutered without being previously assigned to a group, they are treated as a group of size one.

Reply:
This resolves an issue that results from the use of `PostOrderVisit` [here](https://github.com/apache/tvm/blob/main/src/relax/transform/fuse_ops.cc#L230), in `VisitUnsupportedNode`.  It looks like this function is intended to collect all `Var`/`Constant` nodes that are part of the RHS of a binding (e.g. extracting `A` and `B` from `C = R.add(A,B)`).  However, it doesn't collect any bindings that may be part of a nested expression.  As a result, variable bindings that are part of a conditional's body would erroneously trigger the `LOG(FATAL)` in `VisitLeaf`.